File: OpenCV演算法精解：基於Python與C++/code/unzip/OpenCV算法精解-代码/第4章：对比度增强/4.6/CLAHE.cpp
#include<opencv2/core.hpp>
#include<opencv2/highgui.hpp>
#include<opencv2/imgproc.hpp>
using namespace cv;
int main(int argc, char*argv[])
{
	//输入图像
	Mat src = imread(argv[1], IMREAD_GRAYSCALE);
	if (!src.data)
		return -1;
	//构建 CLAHE 对象
	Ptr<CLAHE> clahe =  createCLAHE(2, Size(8, 8));
	Mat dst;
	//限制对比度的自适应直方图均衡化
	clahe->apply(src, dst);
	//显示原图及均衡化后的效果
	imshow("原图", src);
	imshow("对比度增强", dst);
	imwrite("clahe.jpg", dst);
	waitKey(0);
	return 0;
}

File: OpenCV演算法精解：基於Python與C++/code/unzip/OpenCV算法精解-代码/第4章：对比度增强/4.6/CLAHE.py
# -*- coding: utf-8 -*-
import cv2
import numpy as np
import sys
#主函数
if __name__ =="__main__":
    if len(sys.argv) > 1:
    #第一步：读入图像
        src = cv2.imread(sys.argv[1],cv2.IMREAD_GRAYSCALE)
    else:
        print "Usge:python CLAHE.py imageFile"
    #创建 CLAHE  对象
    clahe = cv2.createCLAHE(clipLimit=1.0,tileGridSize=(28,28))
    #限制对比度的自适应阈值均衡化
    dst = clahe.apply(src)
    #显示
    cv2.imshow("src",src)
    cv2.imshow("clahe",dst)
    cv2.waitKey(0)
    cv2.destroyAllWindows()
